src/components/MessageCard/MessageCard.jsx

```diff
@@ -86,16 +99,45 @@ export class MessageCard extends React.PureComponent {
   renderImage() {
     const { image } = this.props
 
-    return image ? (
+    if (!image) {
+      return null
+    }
+
+    const { height, width } = this.calculateSize(image)
+
+    return (
       <ImageContainerUI>
         <ImageUI
           src={image.url}
           alt={image.altText || 'Message image'}
-          width={image.width || '100%'}
-          height={image.height || 'auto'}
+          width={width || '100%'}
+          height={height || 'auto'}
         />
       </ImageContainerUI>
-    ) : null
+    )
+  }
+
+  // Calculate size of image to keep the original aspect ratio, but fit within 278x278 square for image
+  calculateSize = image => {
+    if (!image.width || !image.height) {
+      return {}
+    }
+    const width = parseInt(image.width)
+    const height = parseInt(image.height)
+
+    // Not necessary to recalculate if it fits within boundaries
+    if (width < MAX_IMAGE_SIZE && height < MAX_IMAGE_SIZE) {
```

Comment:
I do wonder if `MAX_IMAGE_SIZE` should be moved as a prop so we can change it per usage. Default could be 278

cc @tinkertrain

Reply:
It is a good idea @plbabin, but the width of message itself is not customizable, so bigger size might not make sense in this case. Probably only to make it smaller. Do you think maybe make it customizable, but limit to 278 maximum either way?

Reply:
IMO if the need to make it customizable has not been needed until now, I'd leave it as is until it is. It's a simple enough change though, so if any of you know that the use case will be indeed needed feel free to add it!

Reply:
I think we don't have any other cases at the moment that would require us to make it customizable. @Charca maybe you know about any?

Reply:
if it's a fixed width then it's all good with me. I wasn't sure if this component has to be fluid or not. We can always revisit if we need it to have different width